### PR TITLE
Jd/area balance network models

### PR DESCRIPTION
**Pull request by jd-lara**

### CODE REVIEW

File: src/devices_models/devices/AC_branches.jl [hunk 1]
```diff
@@ -430,17 +431,86 @@ function add_constraints!(
...
+    branch_Type::DataType,
+)
+    branch_flow_expr = add_expression_container!(container,
+        PTDFBranchFlow(),
+        branch_Type,
+        branches,
+        time_steps,
+    )
+
+    t1_ = time()
```
Comment (jd-lara) [inline src/devices_models/devices/AC_branches.jl:485]: @daniel-thom this is the approach I took

File: src/core/optimization_container.jl [hunk 2]
```diff
@@ -483,18 +483,146 @@ function _make_system_expressions!(
...
+function _make_system_expressions!(
+    container::OptimizationContainer,
+    subnetworks::Dict{Int, Set{Int}},
+    ::Type{AreaBalancePowerModel},
+    areas::IS.FlattenIteratorWrapper{PSY.Area},
+)
+    if length(subnetworks) > 1
+        throw(
+            IS.ConflictingInputsError(
+                "AreaBalancePowerModel doesn't support systems with multiple asynchrous areas",
```
Comment (rodrigomha) [inline src/core/optimization_container.jl:495]: typo: asynchrous -> asynchronous

File: src/devices_models/devices/AC_branches.jl [hunk 3]
```diff
@@ -430,17 +431,103 @@ function add_constraints!(
...
+    return
+end
+
+const ValidPTDFS = Union{
+    PNM.PTDF{
+        Tuple{Vector{Int}, Vector{String}},
+        Tuple{Dict{Int64, Int64}, Dict{String, Int64}},
+        Matrix{Float64},
+    },
+    VirtualPTDF{
```
Comment (rodrigomha) [inline src/devices_models/devices/AC_branches.jl:443]: This should be `PNM.VirtualPTDF`

File: src/devices_models/devices/AC_branches.jl [hunk 4]
```diff
@@ -457,33 +544,39 @@ function add_constraints!(
...
                 jump_model,
-                sum(
-                    ptdf_col[i] * nodal_balance_expressions.data[i, t] for
-                    i in 1:length(ptdf_col)
-                ) - flow_variables_[t] == 0.0
+                branch_flow_expr[name, t] - flow_variables[name, t] == 0.0
             )
         end
     end
+    t2 = time()
```
Comment (rodrigomha) [inline src/devices_models/devices/AC_branches.jl:566]: same here for `t1_` and `t2_`

File: src/network_models/area_balance_model.jl [hunk 5]
```diff
@@ -1,4 +1,29 @@
-function area_balance(
+function add_constraints(
```
Comment (rodrigomha) [inline src/network_models/area_balance_model.jl:1]: this one does not have a bang !, not sure who's calling this method.

**Pull request merged**